internal/storage/rocksdb/store.go

```diff
@@ -508,34 +652,79 @@ func toByteArray(value *gorocksdb.Slice) []byte {
 	return res
 }
 
+func parseTTLMsgPackData(valueWithTtl []byte) (*ttlDataFormat, error) {
+	var row ttlDataFormat
+	err := msgpack.Unmarshal(valueWithTtl, &row)
+	return &row, err
+}
+
 func (rdb *rocksDB) getSingleKey(ro *gorocksdb.ReadOptions, key []byte) ([]*serverpb.KVPair, error) {
 	defer rdb.opts.statsCli.Timing("rocksdb.single.get.latency.ms", time.Now())
-	value, err := rdb.db.Get(ro, key)
+	values, err := rdb.db.MultiGetCFMultiCF(ro, []*gorocksdb.ColumnFamilyHandle{rdb.normalCF, rdb.ttlCF}, [][]byte{key, key})
 	if err != nil {
 		rdb.opts.statsCli.Incr("rocksdb.single.get.errors", 1)
 		return nil, err
 	}
-	defer value.Free()
-	val := toByteArray(value)
-	if val != nil && len(val) > 0 {
-		return []*serverpb.KVPair{&serverpb.KVPair{Key: key, Value: val}}, nil
+	value1, value2 := values[0], values[1]
+	kv := rdb.extractResult(value1, value2, key)
+	if kv != nil {
+		return []*serverpb.KVPair{kv}, nil
 	}
 	return nil, nil
 }
 
+func (rdb *rocksDB) extractResult(value1 *gorocksdb.Slice, value2 *gorocksdb.Slice, key []byte) *serverpb.KVPair {
+	defer value1.Free()
```

Comment:
It might be better to have the caller de-alloc these, since we do not know if anymore ops need to be done there.